File: src/main/java/com/bdev/hogwarts_api/data/converter/student_status/StudentStatusModelConverter.java
package com.bdev.hogwarts_api.data.converter.student_status;

import com.bdev.hogwarts_api.data.dto.student.StudentStatus;
import com.bdev.hogwarts_api.data.model.student_status.StudentStatusModel;

public class StudentStatusModelConverter {
    public static StudentStatus convert(StudentStatusModel studentStatusModel) {
        return StudentStatus.builder()
                .id(studentStatusModel.getId())
                .studentId(studentStatusModel.getStudentId())
                .status(studentStatusModel.getStatus())
                .creationTime(studentStatusModel.getCreationTime())
                .build();
    }
}

File: src/main/java/com/bdev/hogwarts_api/data/converter/student_status/StudentStatusModelConverter.kt
package com.bdev.hogwarts_api.data.converter.student_status

import com.bdev.hogwarts_api.data.dto.student.StudentStatus
import com.bdev.hogwarts_api.data.model.student_status.StudentStatusModel

object StudentStatusModelConverter {
    fun convert(studentStatusModel: StudentStatusModel): StudentStatus {
        return StudentStatus(
                id = studentStatusModel.id,
                studentId = studentStatusModel.studentId ?: throw RuntimeException(),
                status = studentStatusModel.status,
                creationTime = studentStatusModel.creationTime,
                actionTime = studentStatusModel.actionTime
        )
    }
}
